### ISSUE TITLE
Issue Exporting DB to Json

### ISSUE THREAD

**Issue by samirame** [jepiqueau/capacitor-sqlite#21]
Hi!

I am using Sqlite plugin  v. 2.1.0-3 and have some difficulties exporting my DB in '**full**' mode to Json.

The scheme of the DB is defined in normal way, as follows:

```
        const sqlcmd = `
        BEGIN TRANSACTION;
        CREATE TABLE IF NOT EXISTS table1 (
          table1id INTEGER PRIMARY KEY NOT NULL,
            col1 TEXT UNIQUE NOT NULL
        );
      CREATE TABLE IF NOT EXISTS table2 (
        table2id INTEGER PRIMARY KEY NOT NULL,
        col1 TEXT UNIQUE NOT NULL
    );
        CREATE TABLE IF NOT EXISTS table3 (
            table3id INTEGER PRIMARY KEY NOT NULL,
            table1id INTEGER NOT NULL,  
            table2id INTEGER DEFAULT 1,    
            col1 TEXT NOT NULL,
            col2	TEXT,
            FOREIGN KEY(table1id) REFERENCES table1(table1id) ON DELETE CASCADE,
            FOREIGN KEY(table2id) REFERENCES table2(table2id) ON DELETE SET DEFAULT
        );

        CREATE INDEX IF NOT EXISTS table3_index_table1_fk ON table3(table1id);
        CREATE INDEX IF NOT EXISTS table3_index_table2_fk ON table3(table2id);
        CREATE INDEX IF NOT EXISTS table3_index_col1 ON table3(col1);
        CREATE INDEX IF NOT EXISTS table1_index_col1 ON table1(col1);
        PRAGMA user_version = 1;
        PRAGMA foreign_keys = ON;
        COMMIT TRANSACTION;`;

```
Then I am using the following code to export the Db to json (similar to the code used in sqlite-starter project):

```
        let result: any = await this.sqliteService.openDB(db_name);
        if (result.result) {
            let retExport = true;
            const expResult: any = await this.sqliteService.exportToJson('full');
            console.log('export result: ', expResult);
            if (Object.keys(result.export).length === 0) {  retExport = false; }
            const jsObj: string = JSON.stringify(expResult.export);
            const jsValidRes = await this.sqliteService.isJsonValid(jsObj);
            if (!jsValidRes.result) { retExport = false; }
            result = await this.sqliteService.close(environment.db_name);
...
```
However, by executing the exportToJson() method, I receive the following response:

` {export: {}, message: "exportToJson: return Obj is not a JsonSQLite Obj"}`

what could cause the problem? is there any restriction/prerequisite to use the exportToJson() function? is it ok to define the table scheme in normal way (i.e. not using Json import to create DB tables)? is it fine to have FOREIGN KEY constraints, DEFAULT and ON DELETE clauses etc. in the DB scheme, or they are not supported by exportToJson('full')?

Any advice would be greatly appreciated!

**Comment by jepiqueau** [comment]
@samirame see my answer to issue#20. FOREIGN KEY are not yet supported in importFromJson and exportToJson, this is the reason why you get this error. I will look at it tomorrow for my understanding what means ON DELETE CASCADE and ON DELETE SET DEFAULT. I will keep you aware of the progress thanks for your patience

**Comment by jepiqueau** [comment]
@samirane Hopefully, i will fix this issue tomorrow, i have the electron & ios plugins ready but i have still to validate the android. so be patient...

**Comment by jepiqueau** [comment]
@samirame This is now fixed in release capacitor-sqlite@2.1.0-4

**Comment by jepiqueau** [comment]
@samirame Come back to me if something wrong.

**Comment by samirame** [comment]
Thank you very much for the new release! The export works fine now 👍 

just for the sake of documentation and with the hope that it might save someone else's time, I would like to mention this experience: 

After updating the plugin to the newest version 2.1.0-4, I still kept getting the error. After few hours of struggling, I found out that in one of the CREATE TABLE statements, I have used "tab", instead of "space", to separate the column name and column type. Replacing tap with space in the sql statement, solved the issue.
